src/docfx/restore/Restore.cs

```diff
@@ -52,6 +52,22 @@ public static (string restoreDir, string url, string refSpec) GetGitRestoreInfo(
             return (PathUtility.NormalizeFolder(dir), url, refSpec);
         }
 
+        /// <summary>
+        /// Get the restore root dir, default is the user proflie dir.
+        /// User can set the DOCFX_APPDATA_PATH environment to change the root
+        /// </summary>
+        public static string GetRestoreRoot()
+        {
+            // TODO: document this environment variable and show it in welcome message
+            var docfxAppData = Environment.GetEnvironmentVariable("DOCFX_APPDATA_PATH");
```

Comment:
this also affects lock, user profile cache. add a static `AppDataPath` to manage them all.